lib/AST/USRGeneration.cpp

```diff
@@ -64,7 +69,9 @@ static bool printObjCUSRFragment(const ValueDecl *D, StringRef ObjCName,
   } else if (isa<AbstractFunctionDecl>(D)) {
     clang::index::generateUSRForObjCMethod(ObjCName, D->isInstanceMember(), OS);
   } else if (isa<EnumDecl>(D)) {
-    OS << "@E@" << ObjCName; // FIXME: expose clang API to handle enum names
+    // FIXME: expose clang API to handle enum names
+    OS << "@M@" << ModuleName;
+    OS << "@E@" << ObjCName;
   } else if (isa<EnumElementDecl>(D)) {
     OS << "@" << ObjCName;
```

Comment:
Ah, I didn't realize that circled back through this code; I thought the "@M" would just be omitted.

It may not be sufficient for extension methods, though, which can be added in a different module than the type being extended.